addons/payment/models/payment_provider.py

```diff
@@ -487,58 +492,84 @@ def _get_compatible_providers(
             *self.env['payment.provider']._check_company_domain(company_id),
             ('state', 'in', ['enabled', 'test']),
         ]
+        providers = self.env['payment.provider'].search(domain)
 
-        # Handle the is_published state.
-        if not self.env.user._is_internal():
-            domain = expression.AND([domain, [('is_published', '=', True)]])
+        # Add payment providers to the report and set the defaults.
+        payment_utils.add_to_report(report, providers)
 
         # Handle the partner country; allow all countries if the list is empty.
         partner = self.env['res.partner'].browse(partner_id)
         if partner.country_id:  # The partner country must either not be set or be supported.
-            domain = expression.AND([
-                domain, [
-                    '|',
-                    ('available_country_ids', '=', False),
-                    ('available_country_ids', 'in', [partner.country_id.id]),
-                ]
-            ])
+            unfiltered_providers = providers
+            providers = providers.filtered(
+                lambda p: not p.available_country_ids or
+                           partner.country_id.id in p.available_country_ids.ids
```

Comment:
```suggestion
                lambda p: not p.available_country_ids or
                          partner.country_id.id in p.available_country_ids.ids
```

Reply:
In fact:
```suggestion
                lambda p: not p.available_country_ids 
                          or partner.country_id.id in p.available_country_ids.ids
```

Reply:
You suggested to reformat the line only for this case but not for the others (like L523 the same file or in the `payment_method.py` L251). Isn't it better to stick to the same format everywhere? Personally I find it clearer like it is, but if there is a coding style that we follow, I will change it everywhere then :)

Reply:
I did it to not make many comments with the same. But it's like when you are doing maths: which one is clearer to you?
```
5,155 +
3 -
119
```
or 
```
5,155
+ 3
- 119
```

Reply:
It depends on the person I guess. But if you insist I will change ofc.